packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1154,10 +1210,31 @@ async function getGCStateFromSnapshot(
         }
         const gcState = await readAndParseBlob<IGarbageCollectionState>(blobId);
         assert(gcState !== undefined, 0x2ad /* "GC blob missing from snapshot" */);
+        // Update latest serialized GC blobs
+        if (trackGCBlobState) {
```

Comment:
The caller skips using the serialized blobs if this option is not set. It will be cleaner if this function shouldn't care about this flag and should always add the blobs and the caller decides what to do with it.